sdks/nodejs/package.json

```diff
@@ -0,0 +1,30 @@
+{
+  "author": "Google Inc.",
+  "bugs": {
+    "url": "https://github.com/GoogleCloudPlatform/agones/issues"
+  },
+  "dependencies": {
+    "google-protobuf": "3.7.0-rc.2",
+    "grpc": "1.18.0"
+  },
+  "description": "SDK for Agones",
+  "devDependencies": {
+    "eslint": "^5.13.0",
+    "jasmine": "^3.3.1",
+    "nyc": "^13.2.0"
+  },
+  "homepage": "https://agones.dev",
+  "license": "Apache-2.0",
+  "main": "src/index.js",
+  "name": "agones",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/GoogleCloudPlatform/agones.git"
+  },
+  "scripts": {
+    "cover": "nyc jasmine",
+    "lint": "eslint --fix src",
+    "test": "jasmine"
+  },
+  "version": "0.8.0"
```

Comment:
We should perhaps think about versioning. E.g. should the SDK version match the agones version? What happens when we release a new version of agones with no SDK change - do we bump the version number and republish to npm? How do we handle patches if a bug is discovered in the SDK?

Reply:
Npm modules are the standard way to install and update SDKs in Node.js.

Publishing should be as simple as executing `npm publish`, so could be automated although you can never unpublish or republish over a version number. Publishing will require a npm account and publish token can be used by the build system if there is a way to keep this secret (this is not something I've automated as we push manually to internal libraries).

Perhaps we can get some pointers from other Google open source Node.js projects?
https://www.npmjs.com/package/googleapis (https://github.com/googleapis/google-api-nodejs-client)

Reply:
Probably first step - let's get some source code that people can use - then we'll look at various options for packaging and installation. https://docs.npmjs.com/cli/install does have a few options there!

Reply:
If a node module is not published to npm then another way to get this is via a git link, e.g `npm install --save GoogleCloudPlatform/agones#release-0.9.0` (or older equivalent `npm install --save git://github.com/GoogleCloudPlatform/agones.git#release-0.9.0` But `package.json` would have to be in the root of the project for that to work. We cannot include the path to the SDK from the repo root as there is no way to install from a git subdirectory: https://github.com/npm/npm/issues/2974

I think in order for someone to use this code they would have to copy at the nodejs sdk path to their own public repository and include that way (what we will do), or copy and paste the code inside their own project. Enough for now?

Reply:
Yeah, I think that is enough for now :+1: people can copy paste for the moment.